### PR TITLE
upgrade Go 1.19 and k8s.io v0.25.0

### PR DESCRIPTION
**Pull request by mfrancisc**
Upgrade Golang and dependencies to match requirements for Openshift 4.12

Jira story: https://issues.redhat.com/browse/CRT-1740

### CODE REVIEW

File: .github/workflows/publish-operators-for-e2e-tests.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
         python-version: '3.x'
 
     - name: Prepare tools
-      uses: codeready-toolchain/toolchain-cicd/prepare-tools-action@master
+      uses: codeready-toolchain/mfrancisc/prepare-tools-action@crt1740
```
Comment (mfrancisc) [inline .github/workflows/publish-operators-for-e2e-tests.yml:67]: temporary, trying to validate the prepare-tools-action

**Comment by mfrancisc** [comment]
I see all the PRs are running `e2e` tests using the `openshift-ci/Dockerfile.tools` file from master branch (so without the updated Golang version). 
For some reason this one in toolchain-e2e is still running go1.17.13 (while other repos are on go1.18.8) and this is causing the job to fail.

I have tried the tests locally and they are passing for me.

If I'm not wrong, I think we will have to bypass branch protection in order to merge this PR.

**Comment by mfrancisc** [comment]
Or I can open a PR and upgrade from go1.17.13 to go1.18.8 , but still will not be using 1.19.7 so not sure how much that makes sense.

**Comment by xcoulon** [comment]
> Or I can open a PR and upgrade from go1.17.13 to go1.18.8 , but still will not be using 1.19.7 so not sure how much that makes sense.

yes, let's try this option

**Review by xcoulon — APPROVED** [review]

**Comment by mfrancisc** [comment]
> yes, let's try this option

@xcoulon please check PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/696

File: go.mod [hunk 2]
```diff
@@ -96,14 +97,18 @@ require (
...
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-go 1.18
+go 1.19
+
+replace github.com/codeready-toolchain/api => github.com/mfrancisc/api v0.0.0-20230403160314-6f2a3be48aea
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/mfrancisc/toolchain-common v0.0.0-20230331125149-52e1310a177b
```
Comment (rajivnathan) [inline go.mod:114]: Friendly reminder

**Review by rajivnathan — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (mfrancisc) [reply go.mod]: fixed in [07e6ac3](https://github.com/codeready-toolchain/toolchain-e2e/pull/694/commits/07e6ac3d51cc70af265d2f7ea511521c5d4926e1)

**Pull request merged**